### ISSUE TITLE
Upgrade Spring.io release-notes generator tool to latest (v0.0.5)

### ISSUE THREAD

**Issue by lfgcampos** [Decathlon/release-notes-generator-action#15]
The action is still using `v0.0.2` as seen here: `ENV RELEASE_NOTE_GENERATOR_VERSION="v0.0.2"`.

Do you guys plan to upgrade to latest (currently at `v0.0.4`)?
I am asking that because this newer version allow us to customize the contributors section.

The `release-notes.yml` structure changed a bit and will break but other than that, I don't think there is any other big change.

**Comment by derTobsch** [comment]
This would be nice. Are there any plans to upgrade or maybe we should just provide a pr.

**Comment by lfgcampos** [comment]
Update on this one: `v0.0.5` is released.

**Comment by lfgcampos** [comment]
Hi @mmornati, thank you for taking time on this.
But I was playing with `v3` but the changes from [this commit](https://github.com/Decathlon/release-notes-generator-action/commit/d090b538d5039ebf735b3f2413f2ad0165a1ba1b) are not in, right?

I guess we should expect a `v3.1` soon with this fix since `v3` is not generating any release-notes anymore.

**Comment by mmornati** [comment]
> Hi @mmornati, thank you for taking time on this.
> But I was playing with `v3` but the changes from [this commit](https://github.com/Decathlon/release-notes-generator-action/commit/d090b538d5039ebf735b3f2413f2ad0165a1ba1b) are not in, right?
> 
> I guess we should expect a `v3.1` soon with this fix since `v3` is not generating any release-notes anymore.

Hey. I tested de V3 without problem on a sample project.
You need to change the yaml configuration file as it is not the same project anymore for spring.

If you have some feedbacks, log or whatever to share I will check them and generate the v 3.0.1 if necessary 😉

**Comment by lfgcampos** [comment]
> > Hi @mmornati, thank you for taking time on this.
> > But I was playing with `v3` but the changes from [this commit](https://github.com/Decathlon/release-notes-generator-action/commit/d090b538d5039ebf735b3f2413f2ad0165a1ba1b) are not in, right?
> > I guess we should expect a `v3.1` soon with this fix since `v3` is not generating any release-notes anymore.
> 
> Hey. I tested de V3 without problem on a sample project.
> You need to change the yaml configuration file as it is not the same project anymore for spring.
> 
> If you have some feedbacks, log or whatever to share I will check them and generate the v 3.0.1 if necessary 😉

I've changed the yaml config as well but my release-notes are empty (only contributors are showing, nothing more).
I was playing with this [here](https://github.com/lfgcampos/gh-actions-playground) (you can see the last actions if needed).

**Comment by mmornati** [comment]
Checking the problem generating the release notes. Sorry @lfgcampos...
I tested it here https://github.com/mmornati/test-relese-notes/runs/1558449868?check_suite_focus=true and all seemed working well.

**Comment by mmornati** [comment]
It seems something related to the configuration file you are using... I put the same and nothing is generated...

**Comment by mmornati** [comment]
After some local tests I found it is related to the configuration with custom labels. The standard ones work but not the custom labels... So I think it is something related to the configuration.
In the end into the changelog project they add lot of stuffs but with a really poor documentation :( 

I will try to find out what is the configuration part which is not working good...
For instance, locally I'm testing in this way:
```
java -jar github-changelog-generator.jar --changelog.repository=mmornati/test-relese-notes --changelog.milestone-reference=id --debug --spring.config.location=.github/release-notes.yml 1 test.md
```

**Comment by lfgcampos** [comment]
> After some local tests I found it is related to the configuration with custom labels. The standard ones work but not the custom labels... So I think it is something related to the configuration.
> In the end into the changelog project they add lot of stuffs but with a really poor documentation :(
> 
> I will try to find out what is the configuration part which is not working good...
> For instance, locally I'm testing in this way:
> 
> ```
> java -jar github-changelog-generator.jar --changelog.repository=mmornati/test-relese-notes --changelog.milestone-reference=id --debug --spring.config.location=.github/release-notes.yml 1 test.md
> ```

Thank you for the effort!
I've cloned their repo and tested locally... and it worked haha
I've justed added `repository: lfgcampos/gh-actions-playground` to the same config file I am using on the other repo and provided the right args to the Application (running inside my IDE): `"Milestone 1.2" "release-notes.md"`

**Comment by mmornati** [comment]
Seems crazy... on my side locally it seems the `--spring.config.location` property is not working at all... or at least, as it is using only default configuration I think it is not getting the value from this parameter.
What is the command you used to run it?

**Comment by mmornati** [comment]
FYI the command run by the action is [here](https://github.com/Decathlon/release-notes-generator-action/blob/d090b538d5039ebf735b3f2413f2ad0165a1ba1b/entrypoint.sh#L61) I'm trying to make it running locally but the`spring.config.location` as it is used by the action is not working anymore.
On the spring documentation, nothing seems changed here... so, no idea :) I'm trying dealing with it to find a solution for the action... but the problem is this one. 
Thanks to point it out !

**Comment by lfgcampos** [comment]
Yes, you might be right... I've placed my `application.yml` under src/main/resources and run it on my IDE. Probably it explains why it works for me but not for you!
Let me do it like you did just to confirm it is not working =)

**Comment by lfgcampos** [comment]
Update: it worked with this command
```
java -jar target/github-changelog-generator.jar --changelog.repository=lfgcampos/gh-actions-playground --changelog.milestone-reference=id --debug --spring.config.location=./release-notes.yml 3 release-notes.md
```

Running from inside their repo folder using their master (I have checked their last commits after `v0.0.5` and I don't see anything that could explain this)

**Comment by mmornati** [comment]
Yeah agreed... it is crazy because outside the repository (I just downloaded the jar on my side) it is not taking the configuration in count.
I open a request on their repository just to know if they have an hint for us :) https://github.com/spring-io/github-changelog-generator/issues/54

On your side, you built the jar by yourself, right?
Can you try with the [officially provided jar](https://github.com/spring-io/github-changelog-generator/releases/download/v0.0.5/github-changelog-generator.jar)? There is maybe a problem inside the last build they made?

**Comment by lfgcampos** [comment]
Well, I have to say it worked with provided jar as well =/

```
java -jar github-changelog-generator.jar --changelog.repository=lfgcampos/gh-actions-playground --changelog.milestone-reference=id --debug --spring.config.location=./release-notes.yml 3 release-notes.md
```

From the output: `:: Spring Boot ::        (v2.3.5.RELEASE)`

I am digging into that as well but thank you very much for the help!

**Comment by lfgcampos** [comment]
Update: if I copy your command line, it does not pick the config as you said... isn't it because of the `.github`? I think it is interpreting this as a file and not a dir or something weird. Try to move the config to the same dir as the jar and it should all work!

**Comment by mmornati** [comment]
Shit... for the actions it will be wired to have the configuration outside the `.github` folder.
We can answer on the issue on the other repository with this information. Really strange... they should make something strange with one of the latest version of spring boot

**Comment by mmornati** [comment]
I can make changes to put configuration file dynamic instead of the static one we are using, so we can temporarly move it outside the .github folder.
I will deliver a new branch in the afternoon to let you test if everything is good on your side too... and we will deliver a 3.0.1 version as you requested.
Sorry about that man but many thanks for your tests and help !!

**Comment by lfgcampos** [comment]
@mmornati no worries, glad to help!
And I am not in a hurry, which means I am fine waiting for them to answer on the issue if you think it is better =)

**Comment by lfgcampos** [comment]
Well, another update... if I try it with their master, it works haha
So I think spring-boot 2.4.1 already have a fix for that (?)

**Comment by mmornati** [comment]
You are the best man. Thanks for your time....
I will try to imagine how to unblock waiting for the new release of the jar.

**Comment by mmornati** [comment]
Found the issue here: https://github.com/spring-projects/spring-boot/issues/23983
You got it :) Great job.

**Comment by mmornati** [comment]
Fixed into the [3.0.1 release](https://github.com/Decathlon/release-notes-generator-action/releases/tag/v3.0.1)

**Comment by lfgcampos** [comment]
> Fixed into the [3.0.1 release](https://github.com/Decathlon/release-notes-generator-action/releases/tag/v3.0.1)

It [worked](https://github.com/lfgcampos/gh-actions-playground/releases/tag/tag-1.2), thank you very much!
